### ISSUE TITLE
PHP 8.5: Warning when using null as an array offset

### ISSUE THREAD

**Issue by MyIgel** [slevomat/coding-standard#1815]
As of PHP 8.5 `Using null as an array offset [...] is now deprecated. Use an empty string instead.`

```
--------------------------------------------------------------------------------
 1 | ERROR | An error occurred during processing; checking has been aborted.
   |       | The error message was: Using null as an array offset is
   |       | deprecated, use an empty string instead in
   |       | /builds/engelsystem/engelsystem/vendor/slevomat/coding-standard/SlevomatCodingStandard/Sniffs/Variables/UnusedVariableSniff.php
   |       | on line 254
   |       | The error originated in the
   |       | SlevomatCodingStandard.Variables.UnusedVariable sniff on line 254.
--------------------------------------------------------------------------------
```

thats in

```php
		} while ($parenthesisOwnerPointer === null && isset($tokens[$actualPointer]['nested_parenthesis']));
```

(above is triggered as an error as i have configured my dev env to do so to find problems faster)

**Comment by MyIgel** [comment]
I am using it on https://github.com/engelsystem/engelsystem with `docker run --rm -u $(id -u):$(id -g) -v $PWD:/mnt --workdir /mnt composer sh -c './vendor/bin/phpcs -p --basepath="$PWD"'`, a smaller "Proof of concept" would be a todo (although this should be enough to test it)

**Comment by MyIgel** [comment]
The `$actualPointer` is set to null after the first iteration for example when testing `tests/Feature/ApplicationFeatureTest.php` in `findOpenerOfNestedParentheses` as `nested_parenthesis` is not set in `$tokens[$pointer]`.

As i don't have enough knowledge in debugging that part further (for example in what case it should be set) i am stuck there with my debugging.  
A possible fix could be checking for `$actualPointer` not being null but i am not sure about possible side-effects?

**Comment by MyIgel** [comment]
Well, its fixed in c9b154a9fc85be43fe99b7b415f931d95359e0c9 so just needed an update to >= [8.22.0](https://github.com/slevomat/coding-standard/releases/tag/8.22.0)...
